felix/bpf/proxy/syncer.go

```diff
@@ -526,6 +526,14 @@ func (s *Syncer) apply(state DPSyncerState) error {
 	log.Infof("Applying new state, %d service", len(state.SvcMap))
 	log.Debugf("Applying new state, %v", state)
 
+	// Extract node labels, service hints annotation and endpoint zone hints
+	// as required to check for Topology Aware Hints implementation on apply.
+	nodeLabels := state.NodeLabels
+	zone, ok := nodeLabels[v1.LabelTopologyZone]
+	if !ok || zone == "" {
+		log.Debugf("Skipping topology aware endpoint filtering since node is missing label '%s'\n", v1.LabelTopologyZone)
+	}
```

Comment:
this is the place where the zone for the node should be extracted. It could be even extracted when the syncer is created. In any case this warning is then checked again in the helper and does not do much here.

Reply:
Thanks @tomastigera - removed accordingly